### PR TITLE
Add docs for header propagation with contour

### PR DESCRIPTION
**Pull request by xtreme-vikram-yadav**
- Fixes request forwarding issues with DomaingMapping and autoTLS enabled

### CODE REVIEW

**Comment by xtreme-vikram-yadav** [comment]
@KauzClay I can't add reviewers but could you please review this since you have context?

File: contour/configuring-contour-with-domainmapping.hbs.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Configure Contour for header propagation with Knative Domain Mapping and autoTLS certs
+
+With autoTLS is enabled, a request for a Knative service going through Domain Mapping may not maintain `x-forwaded-proto` header. This can cause issues when the request is forwarded to a system expecting this header to be set to `https` for example: Spring Security SSO.
+
+This issue can be solved by configuring contour with `num-trusted-hops: 1` which allows the header to correctly propagate via Domain Mapping.
+
+Example contour config map inside TAP cluster:
+
+`kubectl get cm contour -n tanzu-system-ingress -o yaml`:
```
Comment (KauzClay) [inline contour/configuring-contour-with-domainmapping.hbs.md:9]: nit: can we change this section to instead show how someone might set this via tap-values? In tap-values, the contour section has a section like:
```
contour:
  contour:
    configFileContents:
      num-trusted-hops: 1
```

When I looked at this the first time, I thought it was telling me to update this configmap manually.

until I saw your "important" note at the bottom.

I wonder if we can just make it super explicit how to change it in tap-values?

**Review by KauzClay — COMMENTED** [review]
thought: I can't decide where the best place for this doc to live is. Because the problem comes from when you are trying to use a Knative Feature.

Maybe when we get this in, we can link the CNR docs on DomainMappings to this page? I'm not sure what is best, I'm just worried this will be hard to find.

**Comment by xtreme-vikram-yadav** [comment]
Yeah, I agree with your idea to put more details about the setup with DomainMapping and this update in the CNRs docs and have a link to this doc.

**Comment by snneji** [comment]
Hi @xtreme-vikram-yadav, were you finished working on this PR? I might have merged too quickly! If you would like further changes feel free to open another PR.

Is this PR for 1.6 only?

**Comment by xtreme-vikram-yadav** [comment]
@snneji This is is for 1.6 only. Thanks for merging!

**Pull request merged**